cmd/zt_traverser_blob_test.go

```diff
@@ -61,6 +65,989 @@ func TestIsSourceDirWithStub(t *testing.T) {
+
+	//create new blob traverser
+	blobTraverser := newBlobTraverser(blobURL, client, ctx, true, true, func(common.EntityType) {}, false, common.CpkOptions{}, false, false, false, common.EPreservePermissionsOption.None(), false)
+
+	//test method and validate
+	isDir, err := blobTraverser.IsDirectory(true)
+	a.True(isDir)
+	a.Nil(err)
+}
+
+// this test calls Traverse and getPropertiesIfSingleBlob and responds with the BlobUsesCustomerSpecifiedEncryption error
+func TestTraverseGetPropErrorCode(t *testing.T) {
+	a := assert.New(t)
+
+	//create mock server
+	srv, close := mock_server.NewServer(mock_server.WithTransformAllRequestsToTestServerUrl())
+	defer close()
+
+	//initialize variables
+	_, rawURL, blobURL, _, _, credential, err := SetUpVariables()
+	a.Nil(err)
+
+	client, err := blobservice.NewClientWithSharedKeyCredential(rawURL, credential,
+		&blobservice.ClientOptions{
+			ClientOptions: azcore.ClientOptions{
+				Transport: srv, //passing in mock server
+			}})
+	a.Nil(err)
+
+	//the mock server returns "BlobUsesCustomerSpecifiedEncryption"
+	srv.AppendResponse(mock_server.WithStatusCode(409), mock_server.WithHeader("x-ms-error-code", "BlobUsesCustomerSpecifiedEncryption"), mock_server.WithBody([]byte(MockErrorBody("BlobUsesCustomerSpecifiedEncryption"))))
+
+	//create new blob traverser
+	blobTraverser := newBlobTraverser(blobURL, client, ctx, true, true, func(common.EntityType) {}, false, common.CpkOptions{}, false, false, false, common.EPreservePermissionsOption.None(), false)
+
+	//test method and validate
+	seenFiles := make(map[string]StoredObject)
+
+	err = blobTraverser.Traverse(nil, func(storedObject StoredObject) error {
+		seenFiles[storedObject.relativePath] = storedObject
+		return nil
+	}, []ObjectFilter{})
+	a.Equal(map[string]StoredObject{}, seenFiles)
+	a.Equal("this blob uses customer provided encryption keys (CPK). At the moment, AzCopy does not support CPK-encrypted blobs. If you wish to make use of this blob, we recommend using one of the Azure Storage SDKs", err.Error())
+}
+
+// this tests if the Traverse function getPropertiesIfSingleBlob status code is 403 and returns error
+func TestTraverseGetPropStatusCode(t *testing.T) {
+	a := assert.New(t)
+
+	//create mock server
+	srv, close := mock_server.NewServer(mock_server.WithTransformAllRequestsToTestServerUrl())
+	defer close()
+
+	//initialize variables
+	_, rawURL, blobURL, _, _, credential, err := SetUpVariables()
+	a.Nil(err)
+
+	client, err := blobservice.NewClientWithSharedKeyCredential(rawURL, credential,
+		&blobservice.ClientOptions{
+			ClientOptions: azcore.ClientOptions{
+				Transport: srv, //passing in mock server
+			}})
+	a.Nil(err)
+
+	//the mock server responds with a status code error
+	srv.AppendResponse(mock_server.WithStatusCode(403), mock_server.WithHeader("x-ms-error-code", "BlobNotFound"), mock_server.WithBody([]byte(MockErrorBody("BlobNotFound"))))
```

Comment:
nit: BlobNotFound is usually a 404 in http world. If we want to use 403, we should take a look at common error codes for Blob and choose an appropriate one.